crates/commitlog/src/index/indexfile.rs

```diff
@@ -241,7 +241,7 @@ pub struct IndexFile<Key> {
 
 impl<Key: Into<u64> + From<u64>> IndexFile<Key> {
     pub fn open_index_file(path: &OffsetIndexFile) -> io::Result<Self> {
```

Comment:
The only difference between `IndexFile::open_index_file` and `IndexFileMut::open_index_file` is that the latter sets the file's length.

I'm wondering if we should either a) just use `IndexFileMut` from `IndexFile` (because we have `impl AsMut<IndexFileMut> for IndexFile`), or b) use `Mmap` (immutable) to back `IndexFile`

Reply:
May be, ideally it should be b but what we have right now is convienient. I would prefer to use `IndexFile` as it requires explicit call to mutate rather than using `IndexFileMut` always.